packages/compose/src/hooks/use-event/index.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useInsertionEffect, useCallback } from '@wordpress/element';
+
+/**
+ * Any function.
+ */
+export type AnyFunction = ( ...args: any ) => any;
+
+/**
+ * Creates a stable callback function that has access to the latest state and
+ * can be used within event handlers and effect callbacks. Throws when used in
+ * the render phase.
+ *
+ * @param callback The callback function to wrap.
+ *
+ * @example
+ *
+ * ```tsx
+ * function Component(props) {
+ *   const onClick = useEvent(props.onClick);
+ *   React.useEffect(() => {}, [onClick]);
+ * }
+ * ```
+ */
+export default function useEvent< T extends AnyFunction >(
+	/**
+	 * The callback function to wrap.
+	 */
+	callback?: T
+) {
+	const ref = useRef< AnyFunction | undefined >( () => {
+		throw new Error( 'Cannot call an event handler while rendering.' );
```

Comment:
Is the callback guaranteed to be used as an event handler? Should we be more specific that this is about an incorrect `useEvent()` callback invocation?

Reply:
No, this is just a function that "untracks" the callback, for use in events and effects. The naming is standard for this well -established pattern. I agree that the error message might be confusing. This was copied straight from Ariakit :P

Reply:
I know! Was just clarifying. We're on the same page, I will update the error message 👍

Reply:
Updated, let me know if it looks better to you now.

Reply:
It does, thanks 🙌